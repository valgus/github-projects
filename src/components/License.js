import React from 'react';
import PropTypes from 'prop-types';


const License = ({name, checked=false, onChange=f=>f}) => {
    return (
        <div className="license-box">
        <label className="checkbox">
        <input type='checkbox' name={name} checked={checked} id={name} onChange={onChange} />
        {name}
      </label>
      </div>
    )
   }

   License.propTypes = {
    name: PropTypes.string.isRequired,
    checked: PropTypes.bool,
    onChange: PropTypes.func.isRequired,
  }
  
  export default License;