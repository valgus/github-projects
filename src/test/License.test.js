import React from 'react'
import License from '../components/License'
import * as data from './test-data'
import './setup';
import renderer from 'react-test-renderer'


describe('License component', () => {
    const license = data.licenses[0];
    const onchange = jest.fn();
    it('renders correctly',() => { 
        const tree = renderer.create(<License name={license.key} onChange={onchange}/>).toJSON();
        expect(tree).toMatchSnapshot()
       })
})


