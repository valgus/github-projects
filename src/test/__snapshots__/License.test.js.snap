// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`License component renders correctly 1`] = `
<div
  className="license-box"
>
  <label
    className="checkbox"
  >
    <input
      checked={false}
      id="mit"
      name="mit"
      onChange={[MockFunction]}
      type="checkbox"
    />
    mit
  </label>
</div>
`;
